daemon/cluster/swarm.go

```diff
@@ -26,6 +26,11 @@ func (c *Cluster) Init(req types.InitRequest) (string, error) {
 	defer c.controlMutex.Unlock()
 	if c.nr != nil {
 		if req.ForceNewCluster {
+
+			if !c.nr.nodeState.IsManager() {
+				return "", errors.New("This node is not a swarm manager. Worker nodes can't be used to modify cluster state. Please run this command on a manager node or promote the current node to a manager.")
+			}
```

Comment:
I think this should be moved down into the section below where `c.mu` is locked, since I believe that mutex needs to be locked to access `nodeState`.

Also, lint is complaining about the error message, so you may need to work around that by moving the message into a string constant. Is there not already a suitable message we can reuse?

Reply:
ok, I will modify it as soon as possible.